woocommerce/wc-admin
# Try: Register a fake wc-admin homepage to for users without permission

Fixes (possibly) #7061 

Note that this branch is attempting to shift responsibilities to the client for capability checks, however it is impossible to fully remove some of the server-side checks and some of the workarounds in this PR feel less than ideal.

I will open a 2nd PR soon to explore shifting some responsibility back to the server and doing this the more "WordPress way."

Users without the `manage_woocommerce` permission were not able to see any wc-admin registered pages.  This PR:
* Registers wc-admin with the cap check of `read` to allow everyone to access `page=wc-admin` pages.
* Removes said menu item if the user cannot manage WooCommerce

### Screenshots

<img width="190" alt="Screen Shot 2021-07-01 at 4 17 12 PM" src="https://user-images.githubusercontent.com/10561050/124184899-384e6000-da88-11eb-9fc5-95ba8d0a27d0.png">


### Detailed test instructions:

1. Create a user with a capability of `read` and `view_woommerce_reports` and `view_admin_dashboard`
2. Log in with that user
3. Check that you can access Analytics pages, but the WooCommerce menu item and homepage are not available
4. Add the capability `manage_woocommerce`
5. Make sure that the a "WooCommerce" menu item is shown that shows the wc-admin homepage
6. Make sure no regressions have occurred with default admin accounts
